pkg/tool/default.go

```diff
@@ -0,0 +1,106 @@
+package tool
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/openshift/backplane-tools/pkg/source/github"
+	"github.com/openshift/backplane-tools/pkg/utils"
+)
+
+type DefaultTool struct {
+	Tool
+	name       string
+	binaryName string // must be retrieved by BinaryName() func
+}
+
+type GithubTool struct {
+	DefaultTool
+	source             *github.Source
+	versionInLatestTag bool
+}
+
+// toolDir returns this tool's specific directory given the root directory all tools are installed in
+func (t *DefaultTool) toolDir(rootDir string) string {
+	return filepath.Join(rootDir, t.name)
+}
+
+// symlinkPath returns the path to the symlink created by this tool, given the latest directory
+func (t *DefaultTool) symlinkPath(latestDir string) string {
+	return filepath.Join(latestDir, t.BinaryName())
+}
+
+// Name returns the name of the tool
+func (t *DefaultTool) Name() string {
+	return t.name
```

Comment:
Nice of you to also have factorised this method at this level by using some members so it can return different values for the various subclasses :)
Same for the next function.